server/plugin.go

```diff
@@ -145,6 +145,16 @@ func (p *Plugin) UserHasLeftTeam(pluginContext *plugin.Context, tm *model.TeamMe
 }
 
 func (p *Plugin) MessageHasBeenPosted(pluginContext *plugin.Context, post *model.Post) {
+	shouldProcessMessage, err := p.Helpers.ShouldProcessMessage(post, plugin.BotID(p.api.Configurator.GetConfig().BotUserID))
```

Comment:
Since we are using the proxy bot here, it will consider the shouldProcessMessage as if the proxy bot. Looking at the shouldprocessmessage behaviour, this would filter out messages sent directly to the app bot. This logic probably should be handled in the Notify function, or even further down.

Reply:
@larkox , can you detail the scenario? Are there posts created by the apps bot that we want the apps bot to notify the app on? The `shouldProcessMessage` explicitly precludes that bot-to-bot messaging, we can relax it by adding another option, but I'd like to understand our scenario first?

Reply:
After taking a look at the code, without more options, it will be fine. Nevertheless, I still believe it is in base wrong. Here a few examples of what will happen if we were to add a few more options.

If we were to add `AllowBots`:
- App A will think it should process it's own posts, which it shouldn't.
- App A will not think it should process Proxy plugin bot messages.

If we were to add `OnlyBotDMs`:
- App A will ignore messages on its own DM.
- App A will process messages on proxy app bot DM.

Reply:
I am still worried about future changes in this function, since may introduce future confusion, but for the time being, works as expected.